controller/space.go

```diff
@@ -435,13 +435,23 @@ func deleteOpenShiftResource(
 			}
 		}
 	}
+
 	if len(errorsList) != 0 {
 		var errString string
 		for _, err = range errorsList {
 			errString += fmt.Sprintf("%s\n", err)
 		}
 		return errors.NewInternalErrorFromString(errString)
 	}
+
+	//delete pipelines from the space
```

Comment:
I guess it's better to extract delete_pipleline logic out from `deleteOpenShiftResource()` ..

.. because it will make **delete_pipeline test clean** [test_link](https://github.com/fabric8-services/fabric8-wit/pull/2352/files#diff-0996e94a9c00b464ed5e411c5beedf30R19).  These tests has nothing to do with deployment.

It can be refactor like:
```
deleteOpenShiftResource() {
  deleteDeployments()
  deletePipelines()
}
deleteDeployments() {
}
deletePipelines() {
}
```